crates/iceberg-catalog/migrations/20250328131139_fix_sequence_number.sql

```diff
@@ -0,0 +1,25 @@
+DELETE
+FROM table_snapshot t1
+    USING table_snapshot t2
+        JOIN "table" ON t2.table_id = "table".table_id
+WHERE t1.created_at < t2.created_at
+  AND t1.table_id = t2.table_id
+  AND t1.sequence_number = t2.sequence_number
+  AND "table".table_format_version != '1';
+
+-- table fmt version 1 has no sequence number, hence we only enforce the unique constraint via a partial unique index
+CREATE OR REPLACE FUNCTION is_table_format_v2(table_id_param uuid)
```

Comment:
```suggestion
CREATE OR REPLACE FUNCTION is_not_table_format_v1(table_id_param uuid)
```

The function does not what the body describes - it would also return true for table_format_version 3.
I know its an enum, but I would surely forget to update the function.